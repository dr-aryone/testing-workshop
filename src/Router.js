import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";

import ChangeName from "./pages/ChangeName/ChangeName";

import Toggle from "./pages/Toggle/Toggle";

import Counter from "./pages/Counter/Counter";

import Home from "./pages/Home/Home";

import Menu from "./components/Menu/Menu";

const Router = () => {
  return (
    <BrowserRouter>
      <Menu />
      <Switch>
        <Route path="/toggle" component={Toggle} />
        <Route path="/counter" component={Counter} />
        <Route path="/change" component={ChangeName} />
        <Route exact path="/" component={Home} />
      </Switch>
    </BrowserRouter>
  );
};

export default Router;
