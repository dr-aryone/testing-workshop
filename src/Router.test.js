import React from "react";
import ReactDOM from "react-dom";
import { render, fireEvent } from "@testing-library/react";

import Router from "./Router";

describe("Router Component", () => {
  it("Renders without crashing", () => {
    ReactDOM.render(<Router />, document.createElement("div"));

    ReactDOM.unmountComponentAtNode(document.createElement("div"));
  });

  it("Match Snapshot", () => {
    const { firstChild } = render(<Router />);

    expect(firstChild).toMatchSnapshot();
  });

  it("Should go to the Toggle Component", () => {
    const { getByText, getByTestId } = render(<Router />);

    const toggleLink = getByText("Toggle");

    fireEvent.click(toggleLink);

    const toggleContainer = getByTestId("toggleContainer");

    expect(toggleContainer).toBeInTheDocument();
  });

  it("Should go to the Counter Component", () => {
    const { getByText, getByTestId } = render(<Router />);

    const counterLink = getByText("Counter");

    fireEvent.click(counterLink);

    const counterContainer = getByTestId("counterContainer");

    expect(counterContainer).toBeInTheDocument();
  });

  it("Should go to the ChangeName Component", () => {
    const { getByText, getByTestId } = render(<Router />);

    const changeNameLink = getByText("Change Name");

    fireEvent.click(changeNameLink);

    const changeNameContainer = getByTestId("changeNameContainer");

    expect(changeNameContainer).toBeInTheDocument();
  });
});
