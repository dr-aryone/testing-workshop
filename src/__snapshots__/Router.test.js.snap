// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router Component Match Snapshot 1`] = `undefined`;
